packages/govuk-frontend/src/govuk/components/password-input/password-input.puppeteer.test.js

```diff
@@ -0,0 +1,363 @@
+const { render, goTo } = require('@govuk-frontend/helpers/puppeteer')
+const { getExamples } = require('@govuk-frontend/lib/components')
+
+const inputClass = '.govuk-js-password-input-input'
+const buttonClass = '.govuk-js-password-input-toggle'
+const statusClass = '.govuk-password-input__sr-status'
+
+describe('/components/password-input', () => {
+  let examples
+
+  beforeAll(async () => {
+    examples = await getExamples('password-input')
+  })
+
+  describe('/components/password-input/preview', () => {
+    describe('when JavaScript is unavailable or fails', () => {
+      beforeAll(async () => {
+        await page.setJavaScriptEnabled(false)
+      })
+
+      afterAll(async () => {
+        await page.setJavaScriptEnabled(true)
+      })
+
+      it('still renders an unmodified password input', async () => {
+        await render(page, 'password-input', examples.default)
+
+        const inputType = await page.evaluate(
+          (inputClass) =>
+            document.querySelector(inputClass).getAttribute('type'),
+          inputClass
+        )
```

Comment:
**suggestion** When looking at evaluating a function against a specific element, I think we should favour [`page.$eval`](https://pptr.dev/api/puppeteer.page._eval) rather than running `document.querySelector` inside the evaluated function.

It makes what we're evaluating against more visible and limits the function to what's actually happening to that element, so it's a little clearer.